client/sidebar/header/actions/CreateRoom.js

```diff
@@ -76,33 +61,29 @@ const CreateRoom = (props) => {
 
 	const items = useMemo(() => [
 		canCreateChannel && {
-			icon: 'hashtag',
-			name: t('Channel'),
-			qa: 'sidebar-create-channel',
+			label: <MenuItemContent name={t('Channel')} icon='hashtag' />,
 			action: createChannel,
 		},
 		canCreateDirectMessages && {
-			icon: 'team',
-			name: t('Direct_Messages'),
-			qa: 'sidebar-create-dm',
+			label: <MenuItemContent name={t('Direct_Messages')} icon='baloon-arrow-left' />,
 			action: createDirectMessage,
 		},
 		discussionEnabled && canCreateDiscussion && {
-			icon: 'discussion',
-			name: t('Discussion'),
-			qa: 'sidebar-create-discussion',
+			label: <MenuItemContent name={t('Discussion')} icon='discussion' />,
 			action: createDiscussion,
 		},
 	].filter(Boolean), [canCreateChannel, canCreateDirectMessages, canCreateDiscussion, createChannel, createDirectMessage, createDiscussion, discussionEnabled, t]);
 
-	const onClick = useMutableCallback((e) => {
-		if (items.length === 1) {
-			return items[0].action(e);
-		}
-		openPopover(e, items);
-	});
+	const menu = useMemo(() => <Menu
+		key='menu'
+		square
+		mi='x4'
+		icon='edit-rounded'
+		options={items}
+		{...props}
+	/>, [items, props]);
 
-	return showCreate ? <Sidebar.TopBar.Action {...props} icon='edit-rounded' onClick={onClick}/> : null;
+	return showCreate ? <Sidebar.TopBar.Action children={menu}></Sidebar.TopBar.Action> : null;
```

Comment:
Good man, but something happen when i was reviewing that...
The Sidebar.TopBar.Action need an icon 'prop', if undefined raises a warning in console, another warning appears related to the Menu included inside as a children (something with button inside button)... maybe the problems must be fixed into rocket.chat.fuselage. ;)